src/seedu/addressbook/commands/EditCommand.java

```diff
@@ -0,0 +1,75 @@
+package seedu.addressbook.commands;
+
+
+import seedu.addressbook.common.Messages;
+import seedu.addressbook.data.exception.IllegalValueException;
+import seedu.addressbook.data.person.*;
+import seedu.addressbook.data.tag.Tag;
+import seedu.addressbook.data.tag.UniqueTagList;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * Removes a person identified using its last displayed index from the address book
+ * and adds a new person as the last entry in the address book.
+ */
+public class EditCommand extends Command{
+    public static final String COMMAND_WORD = "edit";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Removes a person identified using it's last displayed index from the address book\n"
+            + "and adds a new person as the last entry in the address book."
+            + "Parameters: INDEX NAME [p]p/PHONE [p]e/EMAIL [p]a/ADDRESS  [t/TAG]...\n"
+            + "Example: " + COMMAND_WORD
+            + " 1 John Doe p/98765432 e/johnd@gmail.com a/311, Clementi Ave 2, #02-25 t/friends t/owesMoney";
+
+    public static final String MESSAGE_EDITTED_PERSON_SUCCESS = "Editting result: %1$s";
+    public static final String MESSAGE_DUPLICATE_PERSON = "This person already exists in the address book";
+
+    private final Person toEdit;
+
+    /**
+     * Convenience constructor using raw values.
+     *
+     * @throws IllegalValueException if any of the raw values are invalid
+     */
+    public EditCommand(int targetVisibleIndex, String name,
+                       String phone, boolean isPhonePrivate,
+                       String email, boolean isEmailPrivate,
+                       String address, boolean isAddressPrivate,
+                       Set<String> tags) throws IllegalValueException {
+        super (targetVisibleIndex);
+        final Set<Tag> tagSet = new HashSet<>();
+        for (String tagName : tags) {
+            tagSet.add(new Tag(tagName));
+        }
+        this.toEdit = new Person(
+                new Name(name),
+                new Phone(phone, isPhonePrivate),
+                new Email(email, isEmailPrivate),
+                new Address(address, isAddressPrivate),
+                new UniqueTagList(tagSet)
+        );
+    }
+
+    public ReadOnlyPerson getPerson() {
+        return toEdit;
+    }
+
+    @Override
+    public CommandResult execute() {
+        try {
+            final ReadOnlyPerson target = getTargetPerson();
+            addressBook.addPerson(toEdit);
+            addressBook.removePerson(target);
```

Comment:
Currently it works fine, but beware that the order of the edited person on the list is now changed.